usaspending_api/bulk_download/v2/views.py

```diff
@@ -219,31 +219,64 @@ def create_bulk_download_job(self, json_request, file_name):
                                                 download_job.json_request))
         return download_job
 
-    def post(self, request):
-        """Return all budget function/subfunction titles matching the provided search text"""
-        json_request = request.data
-
+    def validate_request(self, json_request):
+        # Initial Argument Checking
         for required_param in ['award_levels', 'filters']:
             if required_param not in json_request:
                 raise InvalidParameterException('{} parameter not provided'.format(required_param))
-        if json_request['filters'] == {}:
+        if not isinstance(json_request['award_levels'], list):
+            raise InvalidParameterException('Award levels parameter not provided as a list')
+        elif len(json_request['award_levels']) == 0:
+            raise InvalidParameterException('At least one award level is required.')
+        if not isinstance(json_request['filters'], dict):
+            raise InvalidParameterException('Filters parameter not provided as a dict')
+        elif len(json_request['filters']) == 0:
             raise InvalidParameterException('At least one filter is required.')
+        filters = json_request['filters']
         json_request['columns'] = json_request.get('columns', [])
         json_request['file_format'] = json_request.get('file_format', 'csv')
 
         fields_defaults = {
             'award_types': list(award_type_mappings.keys()),
-            'agency': None,
-            'sub_agency': None,
+            'agency': '',
+            'sub_agency': '',
+            'date_range': {},
             'date_type': 'action_date',
-            'keyword': None
+            'keyword': ''
         }
         for field in fields_defaults:
-            json_request['filters'][field] = json_request['filters'].get(field, fields_defaults[field])
-        json_request['filters']['date_range'] = json_request['filters'].get('date_range', {})
-        json_request['filters']['date_range']['start_date'] = (json_request['filters']['date_range']
-                                                               .get('start_date', None))
-        json_request['filters']['date_range']['end_date'] = json_request['filters']['date_range'].get('end_date', None)
+            filters[field] = filters.get(field, fields_defaults[field])
+            if not isinstance(filters[field], type(fields_defaults[field])):
+                type_name = type(fields_defaults[field]).__name__
+                raise InvalidParameterException('{} parameter not provided as a {}'.format(field, type_name))
+        filters['date_range']['start_date'] = filters['date_range'].get('start_date', '')
```

Comment:
is it intentional to overwrite what's in `filters` for the `date_range`? For the date_range key, we should also enforce they provide _something_. It shouldn't be empty since that could fail the checks below for the day difference calculation.

Empty strings would also pass this check `if not isinstance(filters['date_range'][date_range_param], str)` which we probably don't want it to.

Reply:
This will still check the year difference below so it will fail if over 1 year between dates. If not dates provided, assumed user wanted all of time, and then the check for 1 year difference fails as a result.